addons/mail/static/src/discuss/message_pin/common/message_model_patch.js

```diff
@@ -0,0 +1,62 @@
+import { patch } from "@web/core/utils/patch";
+import { Message } from "@mail/core/common/message_model";
+import { Record } from "@mail/core/common/record";
+import { _t } from "@web/core/l10n/translation";
+import { MessageConfirmDialog } from "@mail/core/common/message_confirm_dialog";
+import { Message as MessageComponent } from "@mail/core/common/message";
+
+patch(Message.prototype, {
+    setup() {
+        super.setup();
+        /** @type {luxon.DateTime} */
+        this.pinned_at = Record.attr(undefined, { type: "datetime" });
+    },
+    /**
+     * Prompts the user for confirmation, then sets pinned to true.
+     */
+    pin() {
+        this._store.env.services.dialog.add(MessageConfirmDialog, {
+            confirmText: _t("Yeah, pin it!"),
+            message: this,
+            messageComponent: MessageComponent,
+            prompt: _t("You sure want this message pinned to %(conversation)s forever and ever?", {
+                conversation: this.thread.prefix + this.thread.displayName,
+            }),
+            size: "md",
+            title: _t("Pin It"),
+            onConfirm: () => {
+                this._store.env.services.orm.call(
+                    "discuss.channel",
+                    "set_message_pin",
+                    [this.thread.id],
+                    { message_id: this.id, pinned: true }
+                );
+            },
+        });
+    },
+
+    /**
+     * Prompts the user for confirmation, then sets pinned to false.
+     */
+    unpin() {
```

Comment:
Nitpicking: these comments could take a single line rather than 3:
```js
/** Prompts the user for confirmation, then sets pinned to false  */
```

Less LOCs for comments means less need to scroll to see whole file at a glance, helps readability therefore maintainability of the code. So this seems minor at first but it adds value!

Reply:
These comments are useless, they are just describing what the code does. I removed them.